src/duckling/canvas/tools/entity-creator-tool.ts

```diff
@@ -1,20 +1,33 @@
 import {Injectable} from '@angular/core';
+import {Graphics, DisplayObject} from 'pixi.js';
 
-import {AttributeDefaultService, EntitySystemService, EntityPositionSetService} from '../../entitysystem';
+import {
+    Entity,
+    AttributeDefaultService,
+    EntitySystemService,
+    EntityPositionSetService,
+    EntityBoxService
+} from '../../entitysystem';
 import {Vector} from '../../math';
 import {SelectionService} from '../../selection';
 import {newMergeKey} from '../../state';
 import {BaseTool, CanvasMouseEvent} from './base-tool';
+import {drawRectangle} from '../drawing';
 
 @Injectable()
 export class EntityCreatorTool extends BaseTool {
     constructor(private _attributeDefaultService : AttributeDefaultService,
                 private _entitySystemService : EntitySystemService,
                 private _entityPositionSetService : EntityPositionSetService,
+                private _entityBoxService : EntityBoxService,
                 private _selection : SelectionService) {
         super();
     }
 
+    getDisplayObject(canvasZoom : number) : DisplayObject {
+        return this._buildSelectionBox(canvasZoom);
```

Comment:
It seems weird that the tool is responsible for scaling itself. We scale entity elements for them. We should do the same for the tool.